pkg/frontend/querymiddleware/results_cache.go

```diff
@@ -48,13 +48,15 @@ var (
 
 // ResultsCacheConfig is the config for the results cache.
 type ResultsCacheConfig struct {
-	cache.BackendConfig `yaml:",inline"`
-	Compression         cache.CompressionConfig `yaml:",inline"`
+	cache.BackendConfig      `yaml:",inline"`
+	Compression              cache.CompressionConfig `yaml:",inline"`
+	LowerTTLWithinTimePeriod time.Duration           `yaml:"lower_ttl_within_time_period,omitempty"`
 }
 
 // RegisterFlags registers flags.
 func (cfg *ResultsCacheConfig) RegisterFlags(f *flag.FlagSet) {
 	f.StringVar(&cfg.Backend, "query-frontend.results-cache.backend", "", fmt.Sprintf("Backend for query-frontend results cache, if not empty. Supported values: %s.", supportedResultsCacheBackends))
+	f.DurationVar(&cfg.LowerTTLWithinTimePeriod, "query-frontend.results-cache.lower-ttl-within-time-period", 0, "The query result is set to use a low TTL for the cache entry if the query is within this period from current time.")
```

Comment:
> Have you considered not adding this config at all, and reading the OOO time window setting instead? I feel this setting will make more complicated to use OOO ingestion.

Yes. But since the ooo time window is an ingester config, we were not sure if it a good idea to use it here as well, because what if users do not set ooo time window for query-frontend? But since ooo time window is in limits (without mention of ingesters in the config name anywhere), we can re-use it and add it in docs that the same limit must be supplied for the query-frontend, hence get rid of the additional flag. WDYT?

Reply:
We already have several config options which apply to multiple config. It's fine, as far as documented. The best practice of running Mimir is just using `config.yaml` and passing the same config to all components, so that you don't have to worry about which config applies to which component.

Reply:
As a reference, look at this config option which must to be set on 3 components:
https://github.com/grafana/mimir/pull/2957/files#diff-b6410588e11858b48f519e0416032fc64c292f87fab23498d0d2b98aed9c62a2R1195-R1200